.gitpod.yml

```diff
@@ -0,0 +1,29 @@
+image: gitpod/workspace-full:latest
+
+tasks:
+  - name: Setup
+    init: |
+      make deps
+    command: |
+      gp sync-done setup
+      exit 0
+  - name: Run frontend
+    command: |
+      gp sync-await setup
+      make watch-frontend
+  - name: Run backend
+    command: |
+      gp sync-await setup
+      export GITEA__server__ROOT_URL="https://3000-$GITPOD_WORKSPACE_ID.${GITPOD_WORKSPACE_CLUSTER_HOST}/"
```

Comment:
Any ideas why this variable is not recognized?

Reply:
Pretty sure it should at least propagate through `air`, but no idea right now, I can't get it to work locally either:

```
ROOT_URL="http://localhost:3333" LOCAL_ROOT_URL="http://localhost:3333" GITEA__server__ROOT_URL="http://localhost:3333" ./gitea
```
still listens on ROOT_URL defined in app.ini for some reason.

Reply:
the `GITEA__...` env vars only exist in docker, because of an init script. You'd need to populate the app.ini file instead of using the env vars.

Reply:
Huh, we should really fix that so env vars are completely parsed in the app.

Reply:
well that would need a lot of rework on our config infrastructure

Reply:
I did added `server root_url` to the `app.ini` manually at the beginning, but then the `app.ini` wasn't created with those default values anymore.

Reply:
> well that would need a lot of rework on our config infrastructure

Doesn't seem to much work to me, just do what [environment-to-ini](https://github.com/go-gitea/gitea/blob/main/contrib/environment-to-ini/environment-to-ini.go) does before initizalizing the `setting` module with those derived values.

Reply:
So you would suggest to start a new PR where I move `environment-to-ini` to be part of `/workspace/gitea/modules/setting/setting.go`?

> This would also eliminate the runtime dependency on golang in the container which in itself is kind of madness.

Looks like go is only in the `build-env` image part which is used for compiling. The gitea image itself doesn't seem to have go as far as I can tell.

Reply:
> So you would suggest to start a new PR where I move environment-to-ini to be part of /workspace/gitea/modules/setting/setting.go?

I you're up to it, I'd certainly welcome it, but for this PR, I suppose something like this should suffice:

```
echo -e `[server]\nROOT_URL=http://...`
```

> Looks like go is only in the build-env image part which is used for compiling. The gitea image itself doesn't seem to have go as far as I can tell.

Ah, I see it's doing `go build`, so yes, no actual runtime dependency.

Reply:
changed it. seems to work perfectly now